### PR TITLE
Refactor/machine readable menus

### PR DESCRIPTION
**Pull request by sapphire-arches**
#1620 should be merged first, this has a subset of those changes just to make it work.

### CODE REVIEW

File: docs/menus/oscillator/wave_index.md [hunk 1]
```diff
@@ -0,0 +1,3 @@
+Index into the wavetable.
+
+This is in the standard 0-50 range, but modulation can reach values between the increments presented by this menu item.
```
Comment (m-m-adams) [inline docs/menus/oscillator/wave_index.md:3]: There's also some hacks to make sure that modulation with an LFO can always access the full range of the wavetable with depth 50

File: src/deluge/gui/menu_item/generate/dmenus/compressor.py [hunk 2]
```diff
@@ -0,0 +1,51 @@
+from dmui.dsl import *
+
+threshold = Menu(
+    "audio_compressor::CompParam",
+    "threshold",
+    ["{name}", "{title}", "params::UNPATCHED_COMPRESSOR_THRESHOLD"],
+    "STRING_FOR_THRESHOLD",
```
Comment (m-m-adams) [inline src/deluge/gui/menu_item/generate/dmenus/compressor.py:7]: I think title should be explicitly stated too

File: src/deluge/gui/menu_item/generate/dmui/dsl.py [hunk 3]
```diff
@@ -0,0 +1,314 @@
...
+        system is in a particular state. A value of None (the default)
+        indicates this menu item is always available.
+    """
+
+    def __init__(
+        self,
+        clazz: str,
+        cpp_name: str,
+        arg_template: list[str],
+        title: str,
```
Comment (m-m-adams) [inline src/deluge/gui/menu_item/generate/dmui/dsl.py:100]: Is title required if it's not used in the cpp initialization arguments?

File: src/deluge/gui/menu_item/generate/dmenus/oscillator/sample.py [hunk 4]
```diff
@@ -0,0 +1,87 @@
...
+    ["{name}", "{title}", "params::LOCAL_OSC_A_PITCH_ADJUST"],
+    "STRING_FOR_OSC_TRANSPOSE_MENU_TITLE",
+    "oscillator/sample/transpose.md",
+    name="STRING_FOR_TRANSPOSE",
+)
+
+pitch_speed = Menu(
+    "sample::PitchSpeed",
+    "samplePitchSpeedMenu",
+    ["{name}"],
```
Comment (m-m-adams) [inline src/deluge/gui/menu_item/generate/dmenus/oscillator/sample.py:63]: What's happening in this one, why isn't there a title?

Comment (sapphire-arches) [reply src/deluge/gui/menu_item/generate/dmenus/oscillator/sample.py:63]: I don't know, that's just how the C++ constructor is

Comment (m-m-adams) [reply src/deluge/gui/menu_item/generate/dmenus/oscillator/sample.py:63]: Ah ok, thought it was shenanigans with the dsl

**Comment by m-m-adams** [comment]
Ok my only nit is that I think title should be optional, as is I feel like the syntax where title is the only positional argument is a bit odd

**Review by m-m-adams — APPROVED** [review]

**Pull request merged**